### PR TITLE
feat(top app bar): dense variant

### PR DESCRIPTION
**Pull request by mgr34**
adds missing dense variant. See issue #518. As noted on discord I am having trouble getting screenshot testing to run. But I did add two. A `TopAppBarDense` and a `TopAppBarProminetDense`. Both will be modified further in an additional PR that adds the correct class to `<TopAppBarFixedAdjust>`

---

I signed it.

### CODE REVIEW

File: test/screenshot/top-app-bar/dense.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const TopAppBarDenseScreenshotTest = () => {
+  return (
+    <div className='top-app-bar-container'>
+      <TopAppBar
+        dense
+        title='Miami, FL'
+        navigationIcon={<MaterialIcon
+          icon='menu'
+          onClick={() => console.log('prominent click')}
```
Comment (moog16) [inline test/screenshot/top-app-bar/dense.js]: dense click*

File: test/screenshot/top-app-bar/prominentDense.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+const TopAppBarProminentDenseScreenshotTest = () => {
+  return (
+    <div className='top-app-bar-container'>
+      <TopAppBar
+        prominent
+        dense
+        title='Miami, FL'
+        navigationIcon={<MaterialIcon
+          icon='menu'
+          onClick={() => console.log('prominent click')}
```
Comment (moog16) [inline test/screenshot/top-app-bar/prominentDense.js]: prominent dense

File: test/unit/top-app-bar/index.test.js [hunk 3]
```diff
@@ -63,6 +63,11 @@ test('has correct fixed class', () => {
   assert.isTrue(wrapper.hasClass('mdc-top-app-bar--fixed'));
 });
 
+test('has correct dense class', () => {
```
Comment (moog16) [inline test/unit/top-app-bar/index.test.js]: can you also add one for dense prominent variant?

**Comment by moog16** [comment]
Please add this commit to your PR https://github.com/material-components/material-components-web-react/pull/523/commits/dc69fe1c93f8650efbfa13ce24f3af011466ca3a

**Comment by moog16** [comment]
Also you need to "sign" in a separate commit. The CLA hasn't passed.

File: test/screenshot/top-app-bar/prominentDense.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import TopAppBar from '../../../packages/top-app-bar';
+
+import MaterialIcon from '../../../packages/material-icon';
+import MainTopAppBarContent from './mainContent';
+
+const TopAppBarProminentDenseScreenshotTest = () => {
```
[the remark this answers is not in the document]
Comment (mgr34) [reply test/screenshot/top-app-bar/prominentDense.js]: as instructed I was going to add the fixed adjust in a separate PR.

**Comment by mgr34** [comment]
I signed it

File: test/screenshot/top-app-bar/prominentDense.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (moog16) [reply test/screenshot/top-app-bar/prominentDense.js]: cool cool... we'll also need to update the golden in that PR too

**Comment by moog16** [comment]
tests passing https://github.com/material-components/material-components-web-react/pull/523

**Comment by moog16** [comment]
Woo! Thanks

**Pull request merged**